server/handler/simulate_status.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/palantir/go-baseapp/baseapp"
+	"github.com/palantir/policy-bot/pull"
+	"github.com/rs/zerolog"
+)
+
+// SimulateStatus returns a SimulateAPIStatusResponse object as json representing a simulated run of policybot
+// on a provided pull request. Accepts query params which may modify the result.
+type SimulateStatus struct {
+	Simulate
+}
+
+type SimulateStatusResponse struct {
+	Status      string `json:"status"`
+	Description string `json:"description"`
+}
+
+type APIError struct {
+	Status int    `json:"status"`
+	Error  string `json:"error"`
+}
+
+func (h *SimulateStatus) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := *zerolog.Ctx(ctx)
+	var response SimulateStatusResponse
+
+	owner, repo, number, ok := parsePullParams(r)
+	if !ok {
+		logger.Error().Msg("failed to parse pull request parameters from request")
+		writeAPIError(w, http.StatusBadRequest)
+		return
+	}
+
+	installation, err := h.Installations.GetByOwner(ctx, owner)
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to get installation for org")
+		writeAPIError(w, http.StatusInternalServerError)
+		return
```

Comment:
For any situation where you are returning a 500 error, you could instead return the error directly (using `hatpear.Try` to wrap the handler) and rely on the [default baseapp error handler](https://github.com/palantir/go-baseapp/blob/develop/baseapp/error.go#L62-L65) to produce an appropriate response.